react/components/Page.jsx

```diff
@@ -1,22 +1,23 @@
-var React = require('react');
-var Link = require('react-router').Link;
-var Sidebar = require('./Sidebar.jsx');
+var React = require(`react`);
+var Link = require(`react-router`).Link;
+var Sidebar = require(`./Sidebar.jsx`);
 
 var elements = (function() {
-  var randomstring = require('randomstring');
-  var elements = [];
+  var randomstring = require(`randomstring`);
+  var elems = [];
   var i=100;
+
   while(i--) {
-    elements.push(randomstring.generate());
+    elems.push(randomstring.generate());
   }
-  return elements;
-}());
+  return elems;
```

Comment:
Yeah but in this case `elements` doesn't *exist* yet, so eslint's error is actually wrong. No shadowing actually occurs. Is there a rule that says "ignore shadow warnings when the variable in question is being shadowing during its own initialilsation"?